editor/sidebar/post-visibility/index.js

```diff
@@ -71,6 +74,21 @@ class PostVisibility extends Component {
 		this.setState( { opened: false } );
 	}
 
+	handleBlur( event ) {
+		if ( this.state.opened && event.relatedTarget !== null && ! this.dialog.contains( event.relatedTarget ) ) {
```

Comment:
Apparently Firefox has some issues with `relatedTarget` as well? There's an interesting idea here for determining the related target via a `setTimeout( () => { const relatedTarget = document.activeElement } , 0 )` when the blur occurs.

https://github.com/facebook/react/issues/2011#issuecomment-123487250